drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/action/FieldActionInspector.java

```diff
@@ -68,6 +69,44 @@ private boolean isValueRedundant(final DTCellValue52 other) {
         }
     }
 
+    private boolean valuesAreEqual(DTCellValue52 value, DTCellValue52 that) {
+
+        switch (value.getDataType()) {
+
+            case STRING:
+                return nullSafeEquals(value.getStringValue(), that.getStringValue());
+            case NUMERIC:
+            case NUMERIC_BIGDECIMAL:
+            case NUMERIC_BIGINTEGER:
+            case NUMERIC_DOUBLE:
+            case NUMERIC_FLOAT:
+            case NUMERIC_INTEGER:
+            case NUMERIC_SHORT:
+            case NUMERIC_BYTE:
+            case NUMERIC_LONG:
+                return nullSafeEquals(value.getNumericValue(), that.getNumericValue());
+            case DATE:
+                return nullSafeEquals(value.getDateValue(), that.getDateValue());
+            case BOOLEAN:
+                return nullSafeEquals(value.getBooleanValue(), that.getBooleanValue());
+            default:
+                return false;
+        }
+    }
+
+    private boolean nullSafeEquals(final Object value1,
```

Comment:
I believe this can be replaced with ```Objects.equals(..)``` as I've seen @jomarko do similar changes in my code!

Reply:
The name is a bit misleading. What it does it compares if the value is equal when `valueTypes` are the same.

The reason for this was that equals failed when both had the `valueType` boolean and the value was "true", but another one had `stringValue` as null and other one at "". 

This is the actual bug, the other changes actually fix other bugs that kept on draining memory as the dtable changed.